### ISSUE TITLE
Alarm

### ISSUE THREAD

**Issue by Roger4R** [Blackymas/NSPanel_HA_Blueprint#1766]
### Question

 It is possible to use "Alarmo" instead of the "Alarm" control ?

### Additional Details

I'm doing a test with the "Alarmo" hacs and integration. When I use the "Alarmo" with a numeric pincode, it doesn't arm/disarm. But without password or pincode it works fine.

**Comment by edwardtfn** [comment]
Please make sure "[Allow the device to make Home Assistant service calls](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/docs/alarm.md#allow-the-device-to-make-home-assistant-service-calls)" is set. If the problem persists, please look for some error message in your Home Assistant logs.

**Comment by russilker** [comment]
My NSPanel is allowed to make service calls and I am able to arm the alarm, but disarming with a PIN does not work from the NSPanel (also using Alarmo). It just hangs and no errors appear on the Home Assistant side. Just updated everything to 4.2.4 yesterday, otherwise working fine.

**Comment by edwardtfn** [comment]
Could you please report this as a bug in the issues area?

**Comment by Roger4R** [comment]
Hello, I managed to solve this by copying the code from UE nextion version password screen and putting it in the US nextion version which is still missing and that's why it wasn't working. Could you see this?

**Comment by russilker** [comment]
I'd be willing to try that, but could you clarify where I could find this code and where I need to copy it to? @Roger4R

**Comment by russilker** [comment]
> Could you please report this as a bug in the issues area?

Sure, I opened bug 1774

**Comment by edwardtfn** [comment]
- #1774

**Comment by Roger4R** [comment]
> I'd be willing to try that, but could you clarify where I could find this code and where I need to copy it to? @Roger4R

Of course. Go to advanced/hmi files on github and download nspanel_eu.HMI. Open this with the nextion editor. In the right side menu, click on the keyb_num page and then click on the benter button. Below you can see 2 windows, click on the TouchReleaseEvent tab in the Event window and copy the code. After that, open the US hmi version and leave it the same.

Compile this and in the top menu click on File and again on TFT file output to export the TFT file that esphome uses.

**Comment by Menethoran** [comment]
can it be assumed that this is going to be pushed in an update soon? (ill jump through the hoops if no, but if yes, ill just wait a couple days for this to fix mine :) )

**Comment by edwardtfn** [comment]
You can try to use `dev` until we release, as I believe this is fixed there.

Just change your panel's yaml from `ref: main` to `ref: dev`, and after flashing it (wirelessly is fine) you can upload the TFT using the button in the device's page.

**Comment by Menethoran** [comment]
> You can try to use `dev` until we release, as I believe this is fixed there.
> 
> Just change your panel's yaml from `ref: main` to `ref: dev`, and after flashing it (wirelessly is fine) you can upload the TFT using the button in the device's page.

simply changing the reference to the dev branch seems to break the build process. (see short logs below).
```
INFO ESPHome 2023.12.9
INFO Reading configuration /config/esphome/panel-dsh.yaml...
INFO Updating https://github.com/esphome/esphome.git@pull/6192/head
INFO Detected timezone 'America/New_York'
INFO Generating C++ source...
INFO Compiling app...
Processing panel-dsh (board: esp32dev; framework: espidf; platform: platformio/espressif32@5.4.0)
--------------------------------------------------------------------------------
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
 - framework-espidf @ 3.40405.230623 (4.4.5) 
 - tool-cmake @ 3.16.4 
 - tool-ninja @ 1.7.1 
 - toolchain-esp32ulp @ 2.35.0-20220830 
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
Reading CMake configuration...
Dependency Graph
|-- ArduinoJson @ 6.18.5
|-- noise-c @ 0.1.4
Compiling .pioenvs/panel-dsh/src/esphome/components/nextion/nextion.o
Compiling .pioenvs/panel-dsh/src/esphome/components/nextion/nextion_commands.o
*** [.pioenvs/panel-dsh/src/esphome/components/nextion/nextion_upload_arduino.o] Source `src/esphome/components/nextion/nextion_upload_arduino.cpp' not found, needed by target `.pioenvs/panel-dsh/src/esphome/components/nextion/nextion_upload_arduino.o'.
========================= [FAILED] Took 19.73 seconds =========================
```

**Comment by edwardtfn** [comment]
Please try cleaning your build files.

- #1636

**Comment by edwardtfn** [comment]
We just released a patch with solves this: [v4.2.5 - Celebrating 1000 Stars with Enhanced Functionality and UI Improvements](https://github.com/Blackymas/NSPanel_HA_Blueprint/releases/tag/v4.2.5)
Could you please update your system and let me know if the issue was solved?
